Duplicati/Library/Main/BackendManager.cs

```diff
@@ -1021,21 +1082,48 @@ private void DoGet(FileEntryItem item)
 
                 if (!m_options.SkipFileHashChecks)
                 {
+                    bool reupload = false;
                     if (item.Size >= 0)
                     {
                         if (dataSizeDownloaded != item.Size)
-                            throw new Exception(Strings.Controller.DownloadedFileSizeError(item.RemoteFilename, dataSizeDownloaded, item.Size));
+                        {
+                            var errmsg = Strings.Controller.DownloadedFileSizeError(item.RemoteFilename, dataSizeDownloaded, item.Size);
+                            if (coreDoGetRepair(item, tmpfile))
+                                reupload = true;
+                            else
+                                throw new Exception(errmsg + Strings.Controller.ParityRepairFailed);
+                        }
                     }
                     else
                         item.Size = dataSizeDownloaded;
 
                     if (!string.IsNullOrEmpty(item.Hash))
                     {
                         if (fileHash != item.Hash)
-                            throw new HashMismatchException(Strings.Controller.HashMismatchError(tmpfile, item.Hash, fileHash));
+                        {
+                            var errmsg = Strings.Controller.HashMismatchError(tmpfile, item.Hash, fileHash);
+                            if (coreDoGetRepair(item, tmpfile))
+                                reupload = true;
+                            else
+                                throw new HashMismatchException(errmsg + Strings.Controller.ParityRepairFailed);
+                        }
                     }
                     else
                         item.Hash = fileHash;
+
+                    if (reupload)
+                    {
+                        // TODO(cmpute): ensure this is the correct way to re-upload

```

Comment:
It's quite tricky here to implement re-uploading process with the renaming strategy. Here I just re-encrypt the data and upload again. Any suggestions on improving this will be welcome!